operators/constellation-node-operator/internal/infrastructure/infrastructure.go

```diff
@@ -0,0 +1,11 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+// Package infrastructure contains a controllers that reconcile on cloud provider infrastructure.
+// It is used to create, delete and update the spec of infrastructure-related k8s resources based on the
+// actual state of the infrastructure.
+// It uses polling (with additional triggers) to check the state of the infrastructure.
+package infrastructure
```

Comment:
My idea was making it extra clear what we are reconciling on (external resources vs k8s resources). This controller would also interfere with the other envtests, since it creates (and deletes) resources that the other envtests manually create / destroy for testing purposes and are supposed to reconcile on in peace.
Not sure what's the easiest way to resolve this if we were to place them all together.

Reply:
Will do this late in the review process to make the diff view easier to look at